src/main/java/model/Currency.java

```diff
@@ -0,0 +1,87 @@
+package model;
+
+import java.io.Serializable;
+import javax.persistence.*;
+
+
+/**
+ * The persistent class for the currency database table.
+ * 
+ */
+@Entity
+@NamedQuery(name="Currency.findAll", query="SELECT c FROM Currency c")
+public class Currency implements Serializable {
```

Comment:
Use lombok for getters, setters, constructors, etc..

https://www.baeldung.com/intro-to-project-lombok